config/areas/search/views.php

```diff
@@ -1,10 +1,19 @@
 <?php
 
 use Kirby\Cms\App;
+use Kirby\Panel\Panel;
+use Kirby\Panel\View;
 
 return [
 	'search' => [
 		'pattern' => 'search',
+		'when'    => function (): bool {
+			// checks core and custom search areas
+			$permissions = App::instance()->user()?->role()->permissions()->toArray() ?? [];
+			$searches    = View::searches(Panel::areas(), $permissions);
+
+			return empty($searches) === false;
```

Comment:
@bastianallgeier @afbora I am just a bit concerned how this performance-wise adds to each load. Having another place where all the Panel areas, searches and permissions need to be gathered.

Reply:
Damn, you are right. I didn't even think about that to be honest.

Reply:
No, they aren't. They are loaded with every view reload.

Reply:
No, I didn't mean that. For example, before the lines in this PR run, aren't the permissions already processed and cached? If the methods called in this PR lines are not called before, you are right.

Reply:
Yes, permissions should be cached. But searches and areas aren't

Reply:
OK, so what's your suggestion to disable entirely search view? Or go without disabling like preview state of the PR?

Reply:
For now, I would not work with the when condition here. As a long-term project, we should refactor our Panel classes though to not use static methods and keep all of this in a more reusable way.

Reply:
OK then the PR is ready to merge 👍